pkgs/applications/office/ledger-autosync/default.nix

```diff
@@ -0,0 +1,49 @@
+{ stdenv, python3Packages, fetchFromGitHub, ledger, hledger, useLedger ? true, useHledger ? true }:
+
+python3Packages.buildPythonApplication rec {
+  pname = "ledger-autosync";
+  version = "1.0.0";
+# no tests included in PyPI tarball
+  src = fetchFromGitHub {
+    owner = "egh";
+    repo = "ledger-autosync";
+    rev = "v${version}";
+    sha256 = "1fn32c02idccdmf9906pxn248qc9basjy2kr2g600806k3qvw84a";
+  };
+
+  propagatedBuildInputs = with python3Packages; [  
+    asn1crypto  
+    beautifulsoup4  
+    cffi  
+    cryptography  
+    entrypoints  
+    fuzzywuzzy  
+    idna  
+    jeepney  
+    keyring  
+    lxml  
+    mock  
+    nose  
+    ofxclient  
+    ofxhome  
+    ofxparse  
+    pbr  
+    pycparser  
+    secretstorage  
+    six  
+  ] ++ (if useLedger then [ledger] else []) ++ (if useHledger then [hledger] else []);  
```

Comment:
using
```
++ stdenv.lib.optionals withLedger [ ledger ]
++ stdenv.lib.optionals withHledger [ hledger ];
```

Will do effectively the same thing, it's just you aren't defining the logic and the data values are doing it for you.